nasl/nasl_ssh.c

```diff
@@ -1896,3 +1897,63 @@ nasl_ssh_shell_close (lex_ctxt *lexic)
 
   return NULL;
 }
+
+/*
+ * NASL SFTP
+ */
+
+/**
+ * @brief Check if SFTP is enabled.
+ * @naslfn{sftp_enabled_check}
+ *
+ * @nasluparam
+ *
+ * - An ssh session id.
+ *
+ * @naslret An integer: 0 on success, -1 (SSH_ERROR) on Channel request
+ * subsystem failure. Greather than 0 means an error during SFTP init.
+ *
+ * @param[in] lexic Lexical context of NASL interpreter.
+ */
+tree_cell *
+nasl_sftp_enabled_check (lex_ctxt *lexic)
```

Comment:
No sure about the naming.
Alternative would be `nasl_sftp_is_enabled` and then return 1 on success. 0 on subsystem failure because that basically means that it is not enabled. And negative values on other errors.

Reply:
I prefer the name as it is, and return different values for different errors and gives a hint to feed devs. Errors can happen because wrong session id (nasl error NULL), error creating the the sftp(ssh error, negative) and initializing the sftp connection (sftp error positive).
Changing the name, just a boolean would be expected.